app/components/ui/breadcrumbs_component.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class UI::BreadcrumbsComponent < ViewComponent::Base
+  def initialize(current_page)
+    @breadcrumbs = [
+      {
+        text: current_page[:page].data.fetch('breadcrumb', current_page[:page].data.fetch('title')),
+        path: current_page[:page].request_path
+      }
+    ]
+
+    current_page[:page].parents.each do |page|
```

Comment:
On some pages `.pages` might return `nil` values in the array, for example on: http://localhost:3000/frameworks/rails/guides/broadcasting-from-jobs


```suggestion
    current_page[:page].parents.compact.each do |page|
```

Reply:
I didn't know that one, nice!